examples/mongocxx/bulk_write.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <bsoncxx/builder/stream/document.hpp>
+
+#include <mongocxx/client.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/uri.hpp>
+
+using bsoncxx::builder::stream::close_document;
+using bsoncxx::builder::stream::document;
+using bsoncxx::builder::stream::finalize;
+using bsoncxx::builder::stream::open_document;
+using mongocxx::bulk_write;
+using mongocxx::model::update_one;
+
+int main(int, char**) {
+    mongocxx::instance inst{};
+    mongocxx::client conn{mongocxx::uri{}};
+
+    auto coll = conn["test"]["coll"];
+
+    // @begin: cpp-bulk-write-upsert
+    auto doc1 = document{} << "a" << 1 << finalize;
+    auto doc2 = document{} << "$set" << open_document << "a" << 2 << close_document << finalize;
+
+    update_one update{std::move(doc1), std::move(doc2)};
+    update.upsert(true);
+
+    bulk_write bulk{};
+    bulk.append(update);
```

Comment:
Since this is an example of a bulk operation, can you expand it to include several CRUD operations?  It would be good to show that you can mix updates with inserts or deletes, for instance.